src/render-dom.js

```diff
@@ -117,22 +117,23 @@ function makeEventsSelector(rootEl$, namespace) {
     return rootEl$
       .first()
       .flatMapLatest(rootEl => {
-        if (rootEl.length === 0) {
-          return Rx.Observable.empty()
-        }
         if (!namespace || namespace.length === 0) {
           return fromEvent(rootEl, eventName, useCapture)
         }
         const descendantSelector = namespace.join(` `)
-        if (matchesSelector(rootEl, descendantSelector)) { // is root Element
-          return fromEvent(rootEl, eventName, useCapture)
-        }
         const topSelector = namespace.join(``)
+        const roof = rootEl.parentElement
         return fromEvent(rootEl, eventName, useCapture).filter(ev => {
-          if (matchesSelector(ev.target, descendantSelector) ||
-              matchesSelector(ev.target, topSelector))
-          {
-            return isStrictlyInRootScope(ev.target)
+          for (let el = ev.target; el && el !== roof; el = el.parentElement) {
+            if (!isStrictlyInRootScope(el)) {
+              return false
```

Comment:
I just tried running my app with the fix-event-delegation branch. Everything works fine except that my use case which is described in #80 is not working anymore. 
The problem seems to be the strictlyInRootScope check. But currently I am confused and not even sure why it was working before.
I am going to create a simple reproducible test case.